core/kazoo_ast/src/fs_prop_usage.erl

```
@@ -0,0 +1,648 @@
+
+event_filter_filename() ->
+    filename:join([code:lib_dir('ecallmgr', 'src')
+                  ,"fs_event_filters.hrl"
+                  ]).
+
+-record(usage, {usages = [] %% places the Data is accessed
+               ,data_var_name = 'Props' %% Tracks current var name
+               ,data_var_aliases = [] %% typically when kz_json:set_value is used
+               ,current_module %% what module are we currently in
+               ,functions = [] %% AST functions loaded
+               ,visited = [] %% MFAs visited (to stop recursion)
+               }).
+
+-spec process() -> [{module(), list()}].
+process() ->
+    io:format("processing ecallmgr FreeSWITCH Props usage: "),
+    Usages = [{Module, Usages} ||
+                 Module <- kz_ast_util:app_modules('ecallmgr'),
+                 (Usages = process(Module)) =/= 'undefined'
+             ],
+    io:format(" done~n"),
+    Usages.
+
+-spec process(module()) -> list().
+process(Module) when is_atom(Module) ->
+    U = process_action(Module),
+    io:format("."),
+    ?DEBUG("  usage for ~p: ~p~n", [Module, U]),
+    U.
+
+process_action(Module) ->
+    FArgs = function_args(Module),
+    process_action(Module, FArgs).
+
+process_action(Module, {_F, _As}=FArg) ->
+    process_action(Module, [FArg]);
+process_action(Module, FArgs) ->
+    case lists:foldl(fun({F, Args}, Acc) ->
+                             process_action(Module, F, Args, Acc)
+                     end
+                    ,[]
+                    ,FArgs
+                    )
+    of
+        [] -> 'undefined';
+        Usages -> Usages
+    end.
+
+
+process_action(_M, 'undefined', _As, Acc) -> Acc;
+process_action(Module, Function, Args, Acc) ->
+    #usage{usages=Us} = process_mfa_call(#usage{current_module=Module}
+                                        ,Module, Function, Args
+                                        ),
+    Us ++ Acc.
+
+%% define entry points for modules
```

@fenollp the code will silently skip modules not explicitly handled.

If someone adds code/modules that utilize a new FS header, that will not be communicated to mod_kazoo at startup (which is the point of this PR - to provide the config to mod_kazoo).

I will add some documentation for how to do that.